test/views/SubmissionsViewSpec.scala

```diff
@@ -37,15 +42,26 @@ import views.tags.ViewTest
 @ViewTest
 class SubmissionsViewSpec extends UnitViewSpec with ExportsTestData with Stubs with Injector {
 
-  private val zone: ZoneId = ZoneId.of("UTC")
   private val page = instanceOf[submissions]
+
   private def createView(
     rejectedSubmissions: Paginated[(Submission, Seq[Notification])] = Paginated(Seq.empty, Page(), 0),
     actionSubmissions: Paginated[(Submission, Seq[Notification])] = Paginated(Seq.empty, Page(), 0),
     otherSubmissions: Paginated[(Submission, Seq[Notification])] = Paginated(Seq.empty, Page(), 0)
   ): Html =
     page(SubmissionsPagesElements(rejectedSubmissions, actionSubmissions, otherSubmissions))(request, messages)
 
+  private def createViewForNavigationBannerTest(enableSecureMessaging: Boolean): HtmlFormat.Appendable = {
+    val secureMessagingConfig = mock[SecureMessagingConfig]
+    when(secureMessagingConfig.isSecureMessagingEnabled).thenReturn(enableSecureMessaging)
+
+    val injector = new OverridableInjector(bind[SecureMessagingConfig].toInstance(secureMessagingConfig))
+    val page = injector.instanceOf[submissions]
+    page(SubmissionsPagesElements(Seq.empty)(mock[PaginationConfig]))(request, messages)
+  }
+
+  private val zone: ZoneId = ZoneId.of("UTC")
```

Comment:
I'd expect that you can replace the "old" way of creating the view with the one you did here. Is there a reason why there can't be a single way of creating the view?

Reply:
The difference is that `SubmissionsViewSpec` contains many tests and having a common `createView` you would have and handle a mock in all of them that is not in use. In addition to test the navigation banner you do not need all submission used by the top `createView`.

Reply:
I am not convinced, TBH 😄 I'd rather go the way of having a single view creation method, the mock can have a default return value set as `false` in `beforeEach` method.
Right now there might be navigation banner or not, depending on what is defined in configuration. So effectively you have 2 instances of this config to handle, but one is kind of 'implicit'.